lonely-lemmings/earlyinternet/gifapp/editorviews.py

```diff
@@ -0,0 +1,152 @@
+from base64 import b64decode, b64encode
+from io import BytesIO
+import json
+import os
+from typing import Union
+
+from PIL import Image as PILImage
+from django.conf import settings
+from django.contrib.auth.decorators import login_required
+from django.http import HttpResponse, HttpResponseNotFound, HttpResponseRedirect
+from django.shortcuts import render, redirect
+from django.views.decorators.csrf import csrf_exempt
+
+from .forms import ProjectForm
+from .models import Project, Image
+
+MEDIA_DIR = settings.MEDIA_ROOT
+
+
+@login_required
+def paint(request, project_name=None) -> HttpResponse:
+    context = {
+        "name": project_name
+    }
+    return render(request, "base_draw.html", context)
+
+
+@login_required
+@csrf_exempt
+def parse_save_request(request, project_name=None) -> HttpResponse:
+    """receives POST request of images as bytes, decodes and saves image paths to database"""
+
+    data = json.loads(request.body)
+    images_blob = data['image_BLOB']
+
+    # get project id
+    project = Project.objects.filter(name=project_name, user_id=request.user)[0]
+
+    # delete all images associated with project
+    images = Image.objects.filter(project_id=project)
+    image_path_list = [get_img_path(img) for img in images]
+    images.delete()
+
+    # delete all local images associated with the project
+    if image_path_list:
+        for img_path in image_path_list:
+            os.remove(img_path)
+
+    # write request images to file and associate them with project
+    for i, blob in enumerate(images_blob):
+        im = PILImage.open(BytesIO(b64decode(blob.split(',')[1])))
+        # fill transparent background with solid white
+        white_bg = PILImage.new("RGBA", im.size, "WHITE")
+        white_bg.paste(im, (0, 0), im)
+        image_name = f"{request.user}_{project_name}_{i}.png"
+        image_dir = os.path.join(MEDIA_DIR, image_name)
+        white_bg.save(image_dir)
+        Image.objects.create(project_id=project, image_data=image_name, animation_position=i)
+    return HttpResponse("Saved")
+
+
+def get_img_path(image: Image) -> str:
+    """returns the path of the image given an image"""
+    img_name = image.image_data.url.split("/")[2]
+    return os.path.join(MEDIA_DIR, img_name)
```

Comment:
nice to see annotations and docstrings. 
nice and short function. 
well written and clear. 

an improvement to this would to use pathlib and urlparse from the standard library.